src/main/java/hudson/plugins/gradle/injection/token/ShortLivedTokenClient.java

```diff
@@ -0,0 +1,69 @@
+package hudson.plugins.gradle.injection.token;
+
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.plugins.gradle.injection.DevelocityAccessKey;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class ShortLivedTokenClient {
+
+    private static final RequestBody EMPTY_BODY = RequestBody.create(new byte[]{});
+    private final OkHttpClient httpClient;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShortLivedTokenClient.class);
+
+    public ShortLivedTokenClient() {
+        httpClient = new OkHttpClient().newBuilder()
+            .connectTimeout(10, TimeUnit.SECONDS)
+            .writeTimeout(10, TimeUnit.SECONDS)
+            .readTimeout(30, TimeUnit.SECONDS)
+            .build();
+    }
+
+    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
+    public Optional<DevelocityAccessKey> get(String server, DevelocityAccessKey accessKey, @Nullable Integer expiry) {
+        String url = normalize(server) + "api/auth/token";
+        if (expiry != null) {
+            url = url + "?expiresInHours=" + expiry;
+        }
+
+        Request request = new Request.Builder()
```

Comment:
I recommend implementing some form of a retry mechanism. That way, if the Develocity instance is down briefly, the build has a chance of still working.

We found a retry mechanism to be quite helpful in our own dogfooding of short-lived access tokens.